### DISCUSSION TITLE
为什么在新建useModel中，调用初始useMode('@@initialState') 控制台会报错？但 user 值可以拿到

### DISCUSSION THREAD

**Discussion by caoLei1997** [umijs/umi#10741] [Q&A] [answered]
import { useModel } from '@umijs/max';
import { useAccess } from '@umijs/max';

const useUser = () => {
  // 权限
  const access = useAccess();
  // 用户数据
  const {
    initialState: { user },
  }: any = useModel('@@initialState');

  console.log('user: ', user);

  return {
    user: user || {},
    ...access,
  };
};

export default useUser;

控制台报错信息
 plugin-model: Invoking 'user' model failed: TypeError: Cannot read properties of undefined

文档看了几遍，没有说不能在自定义的useModel中调用初始model，所以想请教一下大佬们，我这样写是不对的吗？
我的本意是不想在页面有太多引入，也用自定义hooks 实现了，但还是对上面的写法有疑惑

**Comment by fz6m** [comment, marked as answer]
你这个 user model 读到空值 undefined 了，确认下 `@@ initialState` 的返回值解构正确了吗？给个最小复现看看。

**Reply by caoLei1997** [reply]
大佬好，非常感谢大佬的提醒，确实是解构那里出了问题。
代码结构修改成下面的格式浏览器报错就消失了。
  const { initialState }: any = useModel('@@initialState') || {};
  const user = initialState?.user || {};

我能不能理解为 useModel 会收集我自定义的 hooks，并在内部初始化，而初始化时由于全局初始状态 @@initialState 中有异步代码可能会是 undefined ，所以会出现错误信息
